### ISSUE TITLE
Allow precland to work indoor

### ISSUE THREAD

**Issue by expntly** [ArduPilot/ardupilot#7192]
#### Issue details
I tried precland (companion board) indoor, with GPS disabled (GPS_TYPE=0, AHRS_GPS_USE=0, EK3_GPS_TYPE=3) and a lidar, but it doesn't pick up on the target.  PL.TAcq shows 0, and further digging seems to indicate this is the problem: https://github.com/ArduPilot/ardupilot/blob/6a94d285de43de976e28bbab4df725c13fd96591/libraries/AC_PrecLand/AC_PrecLand.cpp#L233

https://github.com/ArduPilot/ardupilot/blob/6a94d285de43de976e28bbab4df725c13fd96591/libraries/AP_NavEKF2/AP_NavEKF2_Control.cpp#L490

It would be great to be able to fly indoor for testing, and let the camera override.

@rmackay9 @jschall 

#### Version
Head

#### Platform
[  ] All
[  ] AntennaTracker
[X] Copter
[  ] Plane
[  ] Rover
[  ] Submarine

#### Airframe type
X4

#### Hardware type
Pixhawk2.1

**Comment by OXINARF** [comment]
@jschall can say more, but I think that not using velocity would be a regression and quite likely wouldn't be very useful for indoors.

**Comment by expntly** [comment]
I think it would be useful to test precland indoor.

Is the only way to do this right now using opticalflow and disable GPS?

**Comment by OXINARF** [comment]
Using beacons should also work.

When I said that without velocity information, it won't be very useful for indoors is that it will probably overshoot much more and when inside you typically have less space than outside.

**Comment by expntly** [comment]
Yeah but the idea is that computer vision should provide you with relatively highly accurate information as to what your positioning is.  I'm happy to test it.  Is there a way to make this work though?  I've tried disabling a few things and a bunch of parameters but it looks like I'm not on the right path.

**Comment by fnoop** [comment]
Precland absolutely does work indoors, @patrickpoirier51 has cheated death many times in his garage/mancave to prove it :)

**Comment by patrickpoirier51** [comment]
Yep... welcome to the ThunderDrone
Flying in GPS denied requires a mean to estimate position, as an option  the OpticalFlow can be used for this purpose. I used EK3 because you can set the EKF home position  and work in guided mode as well.

For tracking I used, RPI, ODROID with code inspired by @jschall  and tested @fnoop  vision position. It worked as well with OpenMV and JEVOIS camera systems.

**Comment by peterbarker** [comment]
Closing as "does work".

**Comment by rmackay9** [comment]
I've always interpreted this as meaning, "allowing precision landing target to be used as a position source to the EKF".  I think I'll re-open this because it's a valid enhancement request still I think.

**Comment by patrickpoirier51** [comment]
Something like using  precland message to feed vision_position_estimate ?

**Comment by rmackay9** [comment]
@patrickpoirier51, yes, something like that.  @rishabsingh3003 and I have discussed this as part of his work on precision landing and he's actually already done some experiments and found that at least with IRLock the position estimate is not very good.  Perhaps with another sensor it might be more useful.
